src/d3/hapticplot.d3.spec.ts

```diff
@@ -140,6 +137,101 @@ describe('VR Haptic Plot', () => {
     expect(getIntervalReset(controller)).toEqual(100);
   });
 
+  it('kdtree for an empty set of data points is null', () => {
+    const dataArray = [];
+    hapticplot.init(scene, dataArray);
+    expect(get3DTree(scene, shape)).toEqual(null);
+  });
+
+  it('builds a correct k-d tree for a scene with a single datapoints', () => {
+    const dataArray = [10];
+    hapticplot.init(scene, dataArray);
+    expect(get3DTree(scene, shape)).toEqual({
+      position: new Vector3(0, 1.7, -0.35),
+      left: null,
+      right: null });
+  });
+
+  it('builds a correct k-d tree of 4 datapoints', () => {
+    const dataArray = [10, 15, 20, 25];
+    hapticplot.init(scene, dataArray);
+    expect(get3DTree(scene, shape)).toEqual({
+      position: new Vector3(0.35, 1.56, -0.35),
+      left: {
+        position: new Vector3(0.175, 1.42, -0.35),
+        left: {
+          position: new Vector3(0, 1.28, -0.35),
+          left: null,
+          right: null,
+        },
+        right: null,
+      },
+      right: {
+        position: new Vector3( 0.5249999999999999, 1.7, -0.35),
+        left: null,
+        right: null,
+      },
+    });
+  });
+
+  it('builds a correct k-d tree for a scene with multiple datapoints with the same value', () => {
+    const dataArray = [5, 5, 5, 5];
+    hapticplot.init(scene, dataArray);
+    expect(get3DTree(scene, shape)).toEqual({
+      position: new Vector3(0.35, 1.7, -0.35),
+      left: {
+        position: new Vector3(0.175, 1.7, -0.35),
+        left: {
+          position: new Vector3(0, 1.7, -0.35),
+          left: null,
+          right: null,
+        },
+        right: null,
+      },
+      right: {
+        position: new Vector3(0.5249999999999999, 1.7, -0.35),
+        left: null,
+        right: null,
+      },
+    });
+  });
+
+  it('distance to closest point is null when no points are in the scene', () => {
+    const dataArray = [];
+    hapticplot.init(scene, dataArray);
+    expect(getDistToNearestPoint(controller, scene, shape, new Vector3(0, 0, 0))).toEqual(null);
+  });
+
+  it('correctly identifies the nearest data in a single point scene', () => {
+    const dataArray = [10];
+    hapticplot.init(scene, dataArray);
+    expect(getDistToNearestPoint(controller, scene, shape, new Vector3(0, 0, 0))).toEqual(1.735655495770978);
+  });
+
+  it('correctly identifies the nearest data point when its positon is the same as the controllers', () => {
+    const dataArray = [10];
+    hapticplot.init(scene, dataArray);
+    expect(getDistToNearestPoint(controller, scene, shape, new Vector3(0, 1.7, -0.35))).toEqual(0);
+  });
+
+  it('correctly identifies the nearest data point when each point has the same value', () => {
```

Comment:
which point is considered the nearest in this case? consider adding that detail to your expectation.

Reply:
I only return the closest distance while searching my tree, and use that value as the input to call my closerPoint and closerPointUnwind search functions recursively. I can add logic to make sure the point is also passed forward through these, but this may have an efficiency effect, and I don't need it for my haptic calculations. I'm not sure how to access this in the testing without doing so. It may be worth doing for the generalizability of the function later on though?